nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -1012,7 +1013,7 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         }
     }
 
-    private Request configureRequest(final ProcessContext context, final ProcessSession session, final FlowFile requestFlowFile, URL url) {
+    private Request configureRequest(final ProcessContext context, final ProcessSession session, final FlowFile requestFlowFile, String url) {
```

Comment:
For clarification, passing a `URL` results in double-encoding, whereas passing the raw `String` property does not? How was this working before the implementation of the custom UriUtils?

Reply:
@exceptionfactory Not exactly. Only when creating a URL with the UriUtils (v1) / UrlValidator (v2) is there the double encoding. Before my changes to how the URL was created there was never a double encoding but if the URL is created with the URI seven argument constructor it will always encode "illegal" characters.

Reply:
Thanks for the additional explanation. So this should restore the previous behavior prior to the multiple iterations that resulted from the need to move away from `java.net.URL` construction.

With this background, it would be helpful to add or update a unit test that evaluates the Transit URI on a Provenance Event, as one way to confirm the expected behavior.

Reply:
Would it be okay to check the value of the `invokehttp.request.url ` attribute which the author of the ticket focused on?

Reply:
Yes, that sounds like a good approach.

Reply:
I have actually incorporated both